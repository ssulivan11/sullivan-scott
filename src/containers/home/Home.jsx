import React from 'react'
import PropTypes from 'prop-types'
import { motion } from 'framer-motion'
import style from './home.scss'

const Home = ({ mainHeadingText, subHeadingText, windowSize, isActive, 'data-test': dataTest }) => (
  <div style={{ height: windowSize.height, position: 'relative' }} data-test={dataTest}>
    <div className={`${style.home} ${isActive ? style['home-active'] : style['home-inactive']}`}>
      <div data-test='home-button' className={style['home-headings']}>
        <motion.h1
          className={`${style.heading} ${style['main-heading']}`}
          transition={{ duration: 1 }}
          animate={{ y: 0, opacity: 1 }}
          initial={{ y: -1000, opacity: 0 }}>
          {mainHeadingText}
        </motion.h1>
        <motion.hr
          className={style['home-hr']}
          transition={{ delay: '0.8' }}
          animate={{ x: 0, opacity: 1 }}
          initial={{ x: 1000, opacity: 0 }}
        />
        <motion.h2
          className={`${style.heading} ${style['sub-heading']}`}
          transition={{ duration: 1 }}
          animate={{ y: 0, opacity: 1 }}
          initial={{ y: 1000, opacity: 0 }}>
          {subHeadingText}
        </motion.h2>
      </div>
    </div>
  </div>
)

Home.propTypes = {
  mainHeadingText: PropTypes.string,
  subHeadingText: PropTypes.string,
  windowSize: PropTypes.shape({
    height: PropTypes.number,
    width: PropTypes.number,
  }),
  isActive: PropTypes.bool,
  'data-test': PropTypes.string,
}

export default Home
