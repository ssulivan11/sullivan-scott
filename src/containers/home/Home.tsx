import * as React from 'react'
import { motion } from 'framer-motion'

import * as style from './home.scss'

interface HomeProps {
  windowSize: {
    height: number
    width: number
  }
  mainHeadingText: string
  subHeadingText: string
  active?: string
  'data-test': string
}

const Home: React.FunctionComponent<HomeProps> = ({
  mainHeadingText,
  subHeadingText,
  windowSize,
  active,
  'data-test': dataTest,
}) => (
  <div style={{ height: windowSize.height, position: 'relative' }} data-test={dataTest}>
    <header className={`container ${style.home} ${active === 'home' ? style['home-active'] : style['home-inactive']}`}>
      <div data-test='home-button' className={style['home-headings']}>
        <motion.h1
          id='title1'
          className={`${style.heading} ${style['main-heading']}`}
          transition={{ duration: 1 }}
          animate={{ y: 0, opacity: 1 }}
          initial={{ y: -100, opacity: 0 }}>
          {mainHeadingText}
        </motion.h1>
        <motion.hr
          className={style['home-hr']}
          transition={{ delay: 1 }}
          animate={{ x: 0, opacity: 1 }}
          initial={{ x: 100, opacity: 0 }}
        />
        <motion.h2
          className={`${style.heading} ${style['sub-heading']}`}
          transition={{ duration: 1 }}
          animate={{ y: 0, opacity: 1 }}
          initial={{ y: 100, opacity: 0 }}>
          {subHeadingText}
        </motion.h2>
        <motion.button
          type='button'
          className={style['home-button']}
          transition={{ delay: 2 }}
          animate={{ opacity: 1 }}
          initial={{ opacity: 0 }}
          whileHover={{ scale: 1.1 }}
          whileTap={{ scale: 0.9 }}
          onClick={() =>
            window.scrollTo({
              top: windowSize.height,
              behavior: 'smooth',
            })
          }>
          Learn more
        </motion.button>
      </div>
    </header>
  </div>
)

export default Home
