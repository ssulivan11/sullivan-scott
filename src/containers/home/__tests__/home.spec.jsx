import React from 'react'
import { shallow, mount } from 'enzyme'

import Home from '../Home'

const defaultTestProps = {
  mainHeadingText: 'Main Heading text',
  subHeadingText: 'Subheading text',
  windowSize: {
    height: 500,
    width: 0,
  },
}

const setup = (renderer = mount, props = defaultTestProps) => {
  const component = renderer(<Home {...props} />)
  return {
    getInstance: () => component,
    getHome: () => component.find(Home),
  }
}

let wrapper
describe('<Home />', () => {
  wrapper = setup(shallow)

  it('renders', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  // it('clicks well', () => {
  //   window.scroll = jest.fn()
  //   wrapper.getSectionOneButton().simulate('click')
  //   expect(window.scroll).toHaveBeenCalled()
  // })
})
