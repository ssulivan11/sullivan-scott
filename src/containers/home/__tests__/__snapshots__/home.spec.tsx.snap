// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> active container 1`] = `
Object {
  "debug": [Function],
  "getHome": [Function],
  "getInstance": [Function],
  "getSectionOneButton": [Function],
}
`;

exports[`<Home /> inactive container 1`] = `
Object {
  "debug": [Function],
  "getHome": [Function],
  "getInstance": [Function],
  "getSectionOneButton": [Function],
}
`;
